pkg/types/openstack/validation/platform.go

```diff
@@ -64,14 +67,29 @@ func ValidatePlatform(p *openstack.Platform, fldPath *field.Path, fetcher ValidV
 	// Validate VIP Values
 	if err := validate.IP(p.APIVIP); err != nil {
 		allErrs = append(allErrs, field.Invalid(fldPath.Child("apiVIP"), p.APIVIP, err.Error()))
-	}
-
-	if err := validate.IP(p.IngressVIP); err != nil {
-		allErrs = append(allErrs, field.Invalid(fldPath.Child("ingressVIP"), p.IngressVIP, err.Error()))
+	} else if n != nil {
+		expectedAPIVIP, _ := cidr.Host(&n.MachineCIDR.IPNet, 5)
+		if p.APIVIP != expectedAPIVIP.String() {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("apiVIP"), p.APIVIP, fmt.Errorf("the API VIP is expected to be %s", expectedAPIVIP).Error()))
+		}
 	}
 
 	if err := validate.IP(p.DNSVIP); err != nil {
 		allErrs = append(allErrs, field.Invalid(fldPath.Child("dnsVIP"), p.DNSVIP, err.Error()))
+	} else if n != nil {
+		expectedDNSVIP, _ := cidr.Host(&n.MachineCIDR.IPNet, 6)
```

Comment:
seems like https://github.com/openshift/installer/pull/2140 is making similar change.. in terms of adding validation..